include/openthread/srp_server.h

```diff
@@ -84,11 +84,17 @@ typedef enum
  *
  * Address mode specifies how the address and port number are determined by the SRP server and how this info is
  * published in the Thread Network Data.
+ *
+ * @warning Using the `OT_SRP_SERVER_ADDRESS_MODE_UNICAST_FORCE_ADD` option will make the implementation
+ * non-compliant with the Thread specification. This option is intended for testing and specific demo purposes.
```

Comment:
> for testing and specific demo purposes

Probably need to mention that one use case is the mobile phone, given this is used in https://github.com/openthread/openthread/pull/11305

Reply:
Updated in new push.